NeoML/src/Dnn/Dnn.cpp

```diff
@@ -539,6 +585,10 @@ void CDnn::DisableLearning()
 
 void CDnn::EnableLearning()
 {
+	if(refInfo != nullptr) {
+		NeoAssert((refInfo->sharedDnns).Size() == 1);
```

Comment:
Better add assert with a string line to describe what exactly goes wrong, and how to use it